src/validators/typed_dict.rs

```diff
@@ -284,6 +288,27 @@ impl Validator for TypedDictValidator {
                             },
                             Err(err) => return Err(err),
                         }
+                    } else if field.validate_always {
+                        // maybe change the fallback of default value to a sentinel ?
+                        let default_value = field.default_value(py)?.unwrap_or(Cow::Owned(py.None()));
+                        match field.validator.validate(
+                            py,
+                            default_value.cast_as::<PyAny>(py)?,
+                            &extra,
+                            slots,
+                            recursion_guard,
+                        ) {
+                            Ok(value) => {
+                                output_dict.set_item(&field.name_pystring, value)?;
+                            }
+                            Err(_) => {
+                                errors.push(ValLineError::new_with_loc(
```

Comment:
Humm, I wonder if we should keep the error the same and set a new flag on the error `error_validating_default`?

Might be a bigger faff, but error detail will be useful here.

Alternatively we could raise this as an internal error since it's not the fault of the input data. WDYT?

Reply:
Internal error seems like a good idea! I'll make some tests

Reply:
We could treat [`PydanticValueError`](https://github.com/pydantic/pydantic-core/blob/6d431dc5f78b567f8e7c8975291edc08a7ce2baa/src/errors/value_exception.rs#L12) in a special way and treat it like a validation error, so there's a way to cause a validation error?

This would be someone confusing since it would be different from functions (and therefore custom types), but I think it would be fine if we document it clearly.